art/attacks/evasion/dpatch_robust.py

```diff
@@ -217,6 +227,23 @@ def generate(self, x: np.ndarray, y: Optional[List[Dict[str, np.ndarray]]] = Non
 
                     patch_gradients_old = patch_gradients
 
+            # Write summary
+            if self.summary_writer is not None:  # pragma: no cover
+                x_patched, y_patched, _ = self._augment_images_with_patch(
+                    x, y, self._patch, channels_first=self.estimator.channels_first
+                )
+
+                self.summary_writer.update(
+                    batch_id=0,
```

Comment:
Is batch_id being held at 0 deliberate? What would be required to record per mini-batch data?

Reply:
Hi @GiulioZizzo Good questions. We have set `batch_id=0` because we can only log metrics for the entire batch in`x` because the batch processing loop is nested inside the EoT loop.